src/deploy/functions/checkIam.ts

```diff
@@ -53,7 +53,7 @@ export async function checkServiceAccountIam(projectId: string): Promise<void> {
  */
 export async function checkHttpIam(context: any, options: any, payload: any): Promise<void> {
   const functionsInfo = payload.functions.triggers;
-  const filterGroups = getFilterGroups(options);
+  const filterGroups = context.filters;
```

Comment:
Good catch on the import! Its not the prettiest code flow, but in prepare.ts, we have a line that already does this work: ``` context.filters = getFilterGroups()```. 

I figured it would be nice to avoid redoing this work - however, it does sacrifice some readability, and it could lead to subtle bugs if this code ever gets used elsewhere. I'll make this fall back to getFilterGroups() if context.filters is undefined.